core/src/v2/actix.rs

```diff
@@ -168,6 +173,40 @@ macro_rules! impl_empty({ $($ty:ty),+ } => {
     )+
 });
 
+#[cfg(feature = "actix4")]
+// Workaround for possibility to directly return HttpResponse from handler
+// after actix removed impl Future from HttpResponse.
+// https://github.com/actix/actix-web/pull/2601
+pub struct HttpResponseWrapper(pub HttpResponse);
```

Comment:
Not sure if this should be in paperclip itself, is this a requirement?

Reply:
I can't figure out a way of not using it (it is used now in tests), and it's also a handy wrapper for end-user, so I inserted it here.

Reply:
On the linked PR @robjtede mentions the impl is useless in practice and removed it so just based on that my first stance would probably be for not adding it here if possible.
But if it is needed for us then I'm ok with it - I'll try to take a better look at this over the weekend.

Reply:
[See note in migration docs.](https://github.com/actix/actix-web/blob/master/actix-web/MIGRATION-4.0.md#returning-httpresponse-synchronously)

Is there a particular reason this is required?

Reply:
When I migrate inline handler to the:

```rust
web::to(|| async { HttpResponse::Ok().finish() })
```

(keep in mind that here `web` is a wrapper for original `web` from actix)

then I get 

```
    --> tests/test_app.rs:2503:36
     |
2503 |                         web::get().to(move || {
     |                                    ^^ the trait `Apiv2Operation` is not implemented for `impl futures::Future<Output = [async output]>`
```

I don't know how to impl `Apiv2Operation` for such type.

We have:

(1) `impl Future for ResponseWrapper<...>` and (2) `impl Apiv2Operation for ResponseWrapper<F, ...>`

`impl Apiv2Operation for F, where F::Future: Future<Output = HttpResponse<BoxBody>>` would probably resolve the problem but it will conflict with (2) because of (1).

When I introduced `HttpResponseWrapper` and implement `Responder` for it, it resolves the situation. I don't know why pure `HttpResponse` is not falling into (1) implementation even though it implements Responder too...

I also tried pin-boxing the result and `impl Apiv2Operation for Pin<Box<dyn Future<...>>`, but with no result.

Reply:
any thoughts @wafflespeanut ?

Reply:
@sfisol should those comments above `HttpResponseWrapper` become doc comment?
I'm trying to think what's the best way to document this, maybe we should add a little example on the quick start?

Reply:
If you try to do it without wrapper then you get error

```
expected struct `HttpResponseWrapper`, found struct `HttpResponse
```

so user is directed to the wrapper by compiler, so I just added a minimal docstring example, and I think this will be sufficient. 

I can't figure out though how to write docstring example that will be tested - `actix_web` and `paperclip` crates are not available in docstring, I don't know why.